### PR TITLE
utf8: enhancements for handling of multibyte sequences

### PR DESCRIPTION
**Pull request by edsiper**
Fix and enhance the way UTF-8 bytes are handled when encoded as JSON. Unit test have been fixed too. 

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

**Comment by edsiper** [comment]
@cosmo0920 would you please take a look at this ?

File: src/flb_sds.c [hunk 1]
```diff
@@ -306,6 +310,30 @@ flb_sds_t flb_sds_cat_utf8 (flb_sds_t *sds, const char *str, int str_len)
...
+        }
+    }
+
+    flb_sds_len_set(s, offset);
+    s[head->len] = '\0';
+    return s;
+
+
+
     for (i = 0; i < str_len; i++) {
```
Comment (cosmo0920) [inline src/flb_sds.c:337]: Can we remove the latter part of flb_sds_cat_utf8?
Or just leaving as-is for now to be able to restore easily?

**Review by cosmo0920 — COMMENTED** [review]
I finally reviewed the newly written utf8 decoding codes.
It sounds nice to me.
I found an issue for this PR. The dead code for flb_sds_cat_utf8 is still existing.
Can we remove that part to simplify the code base?

File: src/flb_sds.c (hunk 1 above)
Comment (edsiper) [reply src/flb_sds.c:337]: thanks, I will do the cleanup shortly

**Pull request merged**